AutoRest/AutoRest.Core/Settings.cs

```diff
@@ -305,10 +320,30 @@ public static void PopulateSettings(object entityToPopulate, IDictionary<string,
                     {
                         try
                         {
-                            if (setting.Value.IsNullOrEmpty() && property.PropertyType == typeof(bool))
+                            if (setting.Value.ToString().IsNullOrEmpty() && property.PropertyType == typeof(bool))
```

Comment:
This removes the null check. If setting.Value is null, ToString() will throw, before the IsNullOrEmpty() call